app/views/teachers/TeacherStudentView.coffee

```diff
@@ -0,0 +1,376 @@
+
+    super(options)
+
+  onLoaded: ->
+    if @students.loaded and not @destroyed
+      @user = _.find(@students.models, (s)=> s.id is @studentID)
+      @updateLastPlayedString()
+      @updateLevelProgressMap()
+      @updateLevelDataMap()
+      @calculateStandardDev()
+      @render()
+    super()
+
+  afterRender: ->
+    super(arguments...)
+    $('.progress-dot, .btn-view-project-level').each (i, el) ->
+      dot = $(el)
+      dot.tooltip({
+        html: true
+        container: dot
+      }).delegate '.tooltip', 'mousemove', ->
+        dot.tooltip('hide')
+
+    @drawBarGraph()
+    @onChangeCourseChart()
+
+
+  onChangeCourseChart: (e)->
+    if (e)
+      selected = ('#visualisation-'+((e.currentTarget).value))
+      $("[id|='visualisation']").hide()
+      $(selected).show()
+
+  calculateStandardDev: ->
+    return unless @courses.loaded and @levels.loaded and @sessions?.loaded and @levelData
+
+    @courseComparisonMap = []
+    for versionedCourse in @classroom.get('courses') or []
+      # course = _.find @courses.models, (c) => c.id is versionedCourse._id
+      course = @courses.get(versionedCourse._id)
+      numbers = []
+      studentCourseTotal = 0
+      members = 0 #this is the COUNT for our standard deviation, number of members who have played all of the levels this student has played.
+      for member in @classroom.get('members')
+        number = 0
+        memberPlayed = 0 # number of levels a member has played that this student has also played
+        for versionedLevel in versionedCourse.levels
+          for session in @sessions.models
+            if session.get('level').original is versionedLevel.original and session.get('creator') is member
+              playedLevel = _.findWhere(@levelData, {levelID: session.get('level').original})
+              if playedLevel.levelProgress is 'complete' or playedLevel.levelProgress is 'started'
+                number += session.get('playtime') or 0
+                memberPlayed += 1
+              if session.get('creator') is @studentID
+                studentCourseTotal += session.get('playtime') or 0
+        if memberPlayed > 0 then members += 1
+        numbers.push number
+
+      # add all numbers[]
+      sum = numbers.reduce (a,b) -> a + b
+
+      # divide by members to get MEAN, remember MEAN is only an average of the members' performance on levels THIS student has done.
+      mean = sum/members
+
+      # # for each number in numbers[], subtract MEAN then SQUARE, add all, then divide by COUNT to get VARIANCE
+      diffSum = numbers.map((num) -> (num-mean)**2).reduce (a,b) -> a+b
+      variance = (diffSum / members)
+
+      # square root of VARIANCE is standardDev
+      StandardDev = Math.sqrt(variance)
+
+      perf = -(studentCourseTotal - mean) / StandardDev
+      perf = if perf > 0 then Math.ceil(perf) else Math.floor(perf)
+
+      @courseComparisonMap.push {
+        courseID: course.get('_id')
+        courseName: course.get('name')
```

Comment:
This (and line 320) should use the translated course names. See: https://github.com/codecombat/codecombat/blob/student-profile-pages/app/templates/courses/teacher-class-view.jade#L255